edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -292,6 +293,22 @@ public Task<IList<ServiceIdentity>> GetDevicesAndModulesInTargetScopeAsync(strin
 
         public async Task<IList<string>> GetAllIds() => await this.serviceIdentityHierarchy.GetAllIds();
 
+        public async Task<SynchedPurchase> GetPurchaseAsync(string deviceId, string moduleId)
```

Comment:
As we discussed it is a bit weird to have purchase status as part of the identity. I have not strong opinion against it, it just feels a bit weird.